### PR TITLE
Add order table to record all services managements

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1723
Related to https://github.com/eclipse-xpanse/xpanse-ui/pull/981

**Deploy service successfully with order record:**
image

**Get status updated or the order is completed with order id:**
image

**Modify service successfully with order record:**
chrome_va30Upqu6n

**Deploy service failed with order record:**
image

**Redeploy failed service with order record:**
image

**Destroy service with order record:**
image

**Purge failed service with order record:**
image

### CODE REVIEW

**Comment by baixinsui** [comment]
@swaroopar I have already tested all order types except for the `migrate`. Do we really need to add an order outside  a workflow instance to record a migration? By the way, please view the commit. I will modify the errors in the test cases next.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOderManager.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.context.request.async.DeferredResult;
+
+/**
+ * Bean to manage service order tasks.
+ */
+@Slf4j
+@Component
+public class ServiceOderManager {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOderManager.java:41]: Typo. Should be ServiceOrderManager

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/polling/ServiceOrderStatusChangePolling.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                new AtomicReference<>(new ServiceOrderStatusUpdate());
+        try {
+            Awaitility.await().atMost(pollingWaitPeriod, TimeUnit.SECONDS)
+                    .pollDelay(0, TimeUnit.SECONDS) // first check runs without wait.
+                    .pollInterval(pollingInterval, TimeUnit.SECONDS).until(() -> {
+                        ServiceOrderEntity serviceOrderEntity = orderStorage.getEntityById(orderId);
+                        TaskStatus taskStatus = serviceOrderEntity.getTaskStatus();
+                        boolean isOrderCompleted = taskStatus == TaskStatus.SUCCESSFUL
+                                || taskStatus == TaskStatus.FAILED;
+                        ServiceOrderStatusUpdate nowStatus = new ServiceOrderStatusUpdate();
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/polling/ServiceOrderStatusChangePolling.java:64]: can we avoid creating this object just for the check? I think we can keep the check like how it was before.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import org.springframework.web.context.request.async.DeferredResult;
+
+/**
+ * Service Orders Management REST API.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_ADMIN, ROLE_USER})
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java:45]: we must also enable this for role ISV . They will need it for service-vendor hosted services.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java [hunk 4]
```diff
@@ -22,41 +23,38 @@
...
+    private UUID orderId;
+
+    @NotNull
+    @Schema(description = "The id of the managed service instance.")
+    private UUID serviceId;
+
     @NotNull
-    @Schema(description = "The state of the service.")
+    @Schema(description = "True if the deployer task is successful.")
+    private Boolean taskIsSuccessful;
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java:36]: variable name should be isTaskSuccessful

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java [hunk 5]
```diff
@@ -92,7 +92,7 @@ public UUID migrate(@Valid @RequestBody MigrateRequest migrateRequest) {
         String userId = getUserId();
         if (!StringUtils.equals(userId, deployServiceEntity.getUserId())) {
             throw new AccessDeniedException(
-                    "No permissions to migrate services belonging to other users.");
+                    "No permissions to migrate service instances belonging to other users.");
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java:95]: the earlier text was fine here.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Comment by swaroopar** [comment]
> @swaroopar I have already tested all order types except for the `migrate`. I did not write code for service migration. Do we really need to add an order outside a workflow instance to record the migration? By the way, please view the commit. I will modify the errors in the test cases next.

@baixinsui Good question. we do not need for migration. as long as we write new order's DEPLOY entry and old order DESTROY entry that should be fine.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    @Tag(name = "ServiceOrders",
+            description = "APIs to manage orders of services")
+    @GetMapping(value = "/services/orders/{orderId}/status",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description =
+            "Long-polling method to get the latest or updated task status of the service order.")
+    @AuditApiRequest(methodName = "getCspFromServiceOrderId")
+    @ResponseStatus(HttpStatus.OK)
+    public DeferredResult<ServiceOrderStatusUpdate> getLatestServiceOrderStatus(
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java:147]: This same method is available in ServiceOrderManageApi also?

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: runtime/src/main/resources/application.properties [hunk 7]
```diff
@@ -14,7 +14,7 @@ spring.datasource.url=jdbc:h2:file:./testdb;DB_CLOSE_ON_EXIT=FALSE
 spring.datasource.driverClassName=org.h2.Driver
 spring.datasource.username=xpanse
 spring.datasource.password=Xpanse@2023
-spring.jpa.show-sql=false
+spring.jpa.show-sql=true
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply runtime/src/main/resources/application.properties:17]: Restored

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java (hunk 6 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java:147]: This new one contains error message is for getting the  task status by the order id. The old one just moved into `ServiceDeployApi` is for getting the  deployment state by the service id.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**